[IV-24-25] Objetivo 2: Modelización del dominio

vpedrosa opened the pull request:
## Sobre la estructura del repositorio

* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas específicas para el lenguaje con el que se está trabajando?

## Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

## Sobre el análisis del problema

* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor, una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo objeto valor que no dependa de ningún otro?

## Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los issues planteados al milestone correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto mínimamente viable?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente correcto?

vpedrosa committed:
feat: implementa objetos valor para condiciones de surf

Añade objetos valor inmutables (WaveHeight, WavePeriod, WaveDirection,
DateRange, SurfConditions) para representar las condiciones de surf de
forma estructurada, garantizando la consistencia de datos en el dominio.

closes #35

vpedrosa committed:
feat: añade entidad SurfSpot con orientación

Implementa la entidad SurfSpot con nombre y orientación para representar
spots de surf específicos y evaluar si las condiciones son apropiadas
según la orientación del spot.

closes #36

vpedrosa committed:
feat: añade entidad ForecastPeriod para análisis temporal

Implementa ForecastPeriod que asocia condiciones de surf a fechas
específicas, permitiendo analizar la evolución temporal de las
condiciones y detectar períodos óptimos.

closes #37

vpedrosa committed:
feat: añade entidad OptimalWindow para resultados del análisis

Implementa OptimalWindow que representa ventanas temporales óptimas
con rango de fechas y puntuación de calidad, permitiendo comunicar
al usuario los mejores períodos para surfear.

closes #38

vpedrosa committed:
feat: añade agregado OptimalPeriodAnalyzer para análisis de períodos

Implementa el agregado raíz OptimalPeriodAnalyzer que coordina el
análisis de pronósticos de un spot para identificar ventanas temporales
óptimas, resolviendo el problema principal de la HU1.

closes #39

vpedrosa committed:
docs: añade configuración y documentación del análisis DDD

Añade archivo iv.yaml con configuración del proyecto, actualiza
.gitignore para TypeScript/Node.js, y documenta el análisis DDD
realizado con la clasificación de entidades, objetos valor y agregados.

vpedrosa committed:
fix: renombra archivos a kebab-case según mejores prácticas

Renombra todos los archivos TypeScript de PascalCase a kebab-case
siguiendo las mejores prácticas para repositorios Node.js/TypeScript.
Actualiza todos los imports y la referencia en iv.yaml.

GabrielFranciscoSM reviewed (commented):
Hola, he estado leyendo tus Issues y el código implementado. ¿No crees que deverías evaluar en una Issue **Con Qué** datos cuentas y **Para Qué** quieres usarlos para desarrollar adecuadamente el resto de problemas / necesidades?

Además, aunque aunque la documentación de la metodología es completa, ¿No crees que conviene justificar mejor cada decisión de implementación en los issues y commits?

vpedrosa commented on docs/ddd-analysis.md:
```diff
@@ -0,0 +1,149 @@
+**Adjetivos (Propiedades):**
+- Óptimo/a
+- Apropiado/a
+- Orientación (suroeste)
```
> aunque la documentación de la metodología es completa, ¿No crees que conviene justificar mejor cada decisión de implementación en los issues y commits?

Sí, estoy de acuerdo. Lo corregiré en los siguientes.

> Hola, he estado leyendo tus Issues y el código implementado. ¿No crees que deverías evaluar en una Issue Con Qué datos cuentas y Para Qué quieres usarlos para desarrollar adecuadamente el resto de problemas / necesidades?

Cierto. Me pongo a ello.

vpedrosa committed:
fix: corrige enlaces rotos en README.md

Actualiza los enlaces a la documentación de historias de usuario y user
journeys para apuntar a los archivos correctos (user-stories.md y
user-journeys.md) en lugar de archivos *-index.md inexistentes.

También corrige el typo "Hitorias" → "Historias".

vpedrosa committed:
docs: documenta propósito de cada dato del dominio

Añade sección "Propósito de los Datos del Dominio" en ddd-analysis.md
que documenta:
- Con qué datos conceptuales contamos (objetos valor y entidades)
- Para qué necesitamos cada dato en la resolución de HU1
- Relación directa entre los datos y los criterios de negocio

Esta documentación responde a las preguntas fundamentales del análisis
DDD: qué datos hemos identificado y por qué los necesitamos.

Closes #41

vpedrosa commented:
@JJ solicito revisión por favor

vpedrosa committed:
build: añade configuración de TypeScript para validar código

Añade package.json y tsconfig.json para permitir la compilación y
validación del código TypeScript del dominio.

Esto cumple el requisito del Objetivo 2: "Código sintácticamente
correcto (compilado/probado)".

- Configuración estricta de TypeScript para mejor calidad de código
- Scripts para compilar (build) y verificar (check) el código
- El código del dominio compila sin errores

Closes #42

JJ commented on docs/ddd-analysis.md:
```diff
@@ -0,0 +1,180 @@
+
+### Palabras Clave Identificadas
+
+Del análisis de las historias de usuario se han identificado los siguientes elementos del dominio:
```
Ese análisis se tiene que convertir en issues, y el código generarse como consecuencia de ese análisis.

vpedrosa replied:
Tienes razón, he invertido el flujo. Los issues los creé después de tener el código, cuando debería haber sido al revés: primero el análisis del dominio genera los issues (como problemas, no como tareas técnicas), y luego el código surge como consecuencia de resolver esos issues.
Sí que había creado issues pero las creé después (están cerradas).

JJ commented on docs/ddd-analysis.md:
```diff
@@ -0,0 +1,180 @@
+   - Inmutable, sin ciclo de vida independiente
+   - Validación: no puede ser negativa
+
+2. **WavePeriod** (`src/domain/value-objects/WavePeriod.ts`)
```
Estos nombres no son los mismos que los qu ehay en el sistema de ficheros. Tampoco pedimos que se haga esto: la documentación tiene que estar en los commits haciendo siempre referencia al issue que se está resolviendo.

JJ commented on src/domain/aggregates/optimal-period-analyzer.ts:
```diff
@@ -0,0 +1,25 @@
+import { OptimalWindow } from "../entities/optimal-window";
+import { SurfSpot } from "../entities/surf-spot";
+
+export class OptimalPeriodAnalyzer {
```
¿Esto es una clase o una función? Si es una función, ¿corresponde a este milestone?

JJ reviewed (commented):
Dado que no se ha hecho ningún issue y el código por tanto no está claro qué problema resuelve, nada del código es válido. Por favor, lee los [errores comunes](https://github.com/JJ/IV-/blob/master/errores/objetivo-2.md) adicionalmente para el nuevo código que incluyas. 
Gracias a @GabrielFranciscoSM por la corrección.

vpedrosa committed:
refactor: elimina código y documentación DDD para reestructurar según metodología

Se elimina el código del dominio y la documentación DDD porque fueron
creados sin seguir el flujo correcto de la metodología:

- Los issues se crearon después del código, no antes
- Los issues estaban planteados como necesidades técnicas, no como
  problemas del dominio derivados de la HU1
- OptimalPeriodAnalyzer contenía lógica de negocio (corresponde a M1)

Se regenerará el código siguiendo el flujo correcto:
1. Análisis del dominio → issues como problemas
2. Código como consecuencia de resolver cada issue

vpedrosa committed:
feat: añade value objects para métricas de condiciones de surf

Resuelve #43

Se crean tres value objects inmutables que representan las métricas
individuales necesarias para evaluar condiciones de surf según la HU1:

- WaveHeight: altura de ola en metros (no negativa)
- WavePeriod: período de pico en segundos (no negativo)
- WaveDirection: dirección del oleaje en grados (0-359)

Cada objeto valida sus restricciones en construcción.

vpedrosa committed:
feat: añade SurfConditions para agrupar métricas de condiciones

Resuelve #44

Se crea SurfConditions como value object inmutable que agrupa las tres
métricas (altura, período, dirección) en una unidad coherente, permitiendo
evaluar de forma conjunta si las condiciones cumplen los criterios de la HU1.

vpedrosa committed:
feat: añade entidad SurfSpot con orientación

Resuelve #45

Se crea la entidad SurfSpot que representa un spot de surf con su nombre
y orientación. La orientación es necesaria para evaluar si la dirección
del oleaje es apropiada para la playa, según los criterios de la HU1.

vpedrosa committed:
feat: añade entidad ForecastPeriod para asociar condiciones a fechas

Resuelve #46

Se crea la entidad ForecastPeriod que asocia condiciones de surf a una
fecha específica. Esto permite analizar las condiciones día a día para
identificar los 35-40 días óptimos al año mencionados en la HU1.

vpedrosa committed:
feat: añade DateRange para representar rangos temporales

Resuelve #47

Se crea el value object DateRange que representa un rango de fechas
(inicio-fin) con validación. Permite delimitar las ventanas temporales
óptimas que el usuario busca identificar según la HU1.

vpedrosa committed:
fix: actualiza iv.yaml para apuntar a entidad existente

La entidad principal es ForecastPeriod, que representa la asociación
entre condiciones de surf y fechas específicas, central para el
análisis de períodos óptimos descrito en la HU1.

vpedrosa committed:
refactor: elimina clases de un solo atributo

Resuelve #48

Según los errores comunes del objetivo 2: "crear una clase con un solo
atributo es algo que habría que replantearse muy seriamente".

Se eliminan WaveHeight, WavePeriod y WaveDirection, moviendo las
validaciones directamente a SurfConditions y SurfSpot.

vpedrosa commented:
@JJ lo he vuelto a hacer (sin cerrar las issues esta vez) y con los commits con la documentación en vez de en un archivo.

JJ reviewed (commented):
En general, en los diferentes envíos vas cayendo en los errores habituales del resto de tus compañeros. Revisar envíos de compañeros tiene ese objetivo; aparte de poder expresar los diferentes conceptos de forma que los pueda entender otra persona (aparte de forma práctica), al ver las revisiones hechas por otros y las que hago yo puedes ver cuales son los errores habituales que, con todo respeto, suele caer la gente.
Por favor, antes de la siguiente ronda, aparte de mirar el fichero de errores habituales, [revisa este PR](https://github.com/gabrielherreraloz/IV-GHL/pull/8) que es el único que queda abierto del objetivo 2 más que con la vista de echarle una mano (que también), con el objeto de aprender de las revisiones que se han hecho. También le pediría a @chelunike que revisara este PR; antes de la siguiente revisión por mi parte me gustaría que él lo hubiera aprobado.
Recuerdo (a ti y a @chelunike) que las revisiones contribuyen a un 20% de la nota final, así que tampoco os viene mal a ninguno de los dos.

JJ commented on package.json:
¿Este fichero lo requiere este milestone?
Aparte que con él (y con tsconfig.json) estás asumiendo un intérprete de JS específico, `node`. Hay otros que no usan uno de los dos ficheros, o los dos.

JJ commented on src/domain/entities/surf-spot.ts:
```diff
@@ -0,0 +1,23 @@
+      throw new Error("Spot name cannot be empty");
+    }
+    if (orientationDegrees < 0 || orientationDegrees >= 360) {
+      throw new Error("Orientation must be between 0 and 359 degrees");
```
Y un error debe incluir *siempre* la causa del mismo para que quien lo reciba entienda cuál es el problema. En este caso, el valor de la orientación. Un error específico, parte del modelo, debería incluir esto aparte del hecho de que sea un error específico para un problema específico, de forma que quien lo reciba pueda tratarlo o "burbujearlo" hasta que llegue al usuario.

vpedrosa replied:
Corregido. Los errores específicos del dominio ahora incluyen el valor que causó el error. Por ejemplo: InvalidWaveDirection incluye readonly value: number y el mensaje muestra el valor incorrecto.

JJ reviewed (commented):
Comentarios adicionales por si pudieran serte útiles.

vpedrosa committed:
refactor: elimina validaciones del modelo

Resuelve #49

Las validaciones son lógica de negocio y corresponden al M1, no al
modelo del dominio. Se eliminan los if/throw y se dejan solo las
estructuras de datos con atributos readonly.

vpedrosa committed:
feat: añade errores específicos del dominio

Resuelve #50

Se crean clases de error específicas para el dominio que incluyen
la causa del error (el valor incorrecto). Estos errores serán
lanzados por la lógica de negocio en M1:

- InvalidWaveHeight: altura de ola negativa
- InvalidWavePeriod: período de ola negativo
- InvalidWaveDirection: dirección fuera de rango 0-359
- InvalidSpotName: nombre de spot vacío
- InvalidDateRange: fecha inicio >= fecha fin

vpedrosa committed:
chore: elimina package.json innecesario

Resuelve #51

El package.json no es requerido por este milestone y asume un
intérprete específico (node). El modelo del dominio no necesita
dependencias de runtime.

vpedrosa commented on src/domain/entities/forecast-period.ts:
```diff
@@ -0,0 +1,19 @@
+    this._conditions = conditions;
+  }
+
+  get date(): Date {
```
> ¿Este milestone, en la descripción de producto mínimo, requiere getters? ¿Lo requiere la historia de usuario? Tampoco tengo claro que este concepto requiera una clase. Si es un argumento que se le va a pasar a la lógica de negocio, basta simplemente con pasar los dos argumentos a la misma, en una función (de otra clase o independiente)

Corregido. He simplificado las clases usando atributos readonly públicos en lugar de getters con propiedades privadas. Sobre si ForecastPeriod necesita ser una clase: la mantengo porque agrupa fecha + condiciones como unidad del dominio, pero entiendo que podría simplificarse más si no aporta valor.

vpedrosa commented:
@chelunike ya he corregido lo que me ha comentado @JJ (gracias, por cierto), y he revisado también tu PR y creo que he corregido todos los errores que tenía. Revísalo cuando puedas, por favor.

chelunike reviewed (commented):
En principio, lo veo bien, por lo menos estamos en el mismo punto. Veo que los issues tienen un buen planteamiento (Cuando mencionas la HU1 #34, sería más cómodo que siempre añadas un enlace). Y por otro lado no termino de ver que los erroes especificos formen parte del domino del problema. Por el resto, no veo ningún otro error.

JJ commented:
> En principio, lo veo bien, por lo menos estamos en el mismo punto. Veo que los issues tienen un buen planteamiento (Cuando mencionas la HU1 #34, sería más cómodo que siempre añadas un enlace). Y por otro lado no termino de ver que los erroes especificos formen parte del domino del problema. Por el resto, no veo ningún otro error.

Los errores específicos *siempre* forman parte del dominio del problema.

vpedrosa commented:
@JJ solicito revisión de nuevo tras la revisión de @chelunike, por favor

JJ commented on src/domain/entities/forecast-period.ts:
```diff
@@ -0,0 +1,11 @@
+import { SurfConditions } from "../value-objects/surf-conditions";
+
+export class ForecastPeriod {
```
Creo que esto ya lo vimos la última vez. ¿Esto está en el dominio del problema o de la solucion? Es decir, ¿es un argumento que le vas a pasar a la lógica de negocio o describe algo que realmente esté en el dominio del problema? Si es parte de la solución no pertenece a este milestone; una vez que se pase al siguiente, esto podrá ser una clase, o simplemente un argumento o dos que se le pasen a la lógica de negocio.

JJ reviewed (commented):
Igual que le has dicho a @chelunike , ¿cuál es la única fuente de verdad para surf-conditions?

vpedrosa committed:
refactor: convierte SurfConditions de Value Object a Entidad #52

SurfConditions representa registros únicos de una boya oceanográfica,
por lo que tiene identidad propia. Se mueve de value-objects/ a entities/
y se añade id y timestamp para garantizar la única fuente de verdad (SSOT).

vpedrosa committed:
refactor: ForecastPeriod referencia SurfConditions por ID

En lugar de contener una copia del objeto SurfConditions, ForecastPeriod
ahora almacena conditionsId como referencia. Esto garantiza que existe
una única fuente de verdad para las condiciones de surf.

Closes #52

vpedrosa committed:
refactor: elimina DateRange del modelo de dominio

DateRange no forma parte del dominio del problema, es un parámetro
de la lógica de negocio que pertenece al M1. Se elimina junto con
su error asociado InvalidDateRange.

Closes #53

vpedrosa committed:
refactor: añade identidad a SurfSpot para garantizar SSOT

SurfSpot representa spots de surf únicos que tienen identidad propia.
Se añade id como identificador único del spot.

Closes #54

vpedrosa commented on src/domain/value-objects/date-range.ts:
```diff
@@ -0,0 +1,9 @@
+export class DateRange {
+  readonly start: Date;
```
Tienes razón, DateRange no pertenece al dominio del problema sino a la solución. Es un argumento que se pasará a la lógica de negocio para filtrar períodos de pronóstico, no un concepto del dominio del surf. Resuelto.

vpedrosa committed:
refactor: elimina ForecastPeriod del modelo de dominio #55

ForecastPeriod no forma parte del dominio del problema, es una
estructura auxiliar de la lógica de negocio que pertenece al M1.

vpedrosa committed:
fix: actualiza iv.yaml para apuntar a SurfConditions

SurfConditions es la entidad fundamental del dominio: representa
los datos de condiciones de surf (altura, período, dirección) que
son el núcleo del problema a resolver.

Closes #55

vpedrosa commented:
@JJ si puedes revisar de nuevo, por favor. Gracias!

oscar0310 commented on src/domain/entities/surf-conditions.ts:
```diff
@@ -0,0 +1,21 @@
+export class SurfConditions {
+  readonly id: string;
```
El identificador de un objeto nunca puede ir dentro de  un objeto.

vpedrosa replied:
He eliminado los identificadores internos de las entidades. El id se gestionará como clave externa en M1.

oscar0310 commented on src/domain/entities/surf-spot.ts:
```diff
@@ -0,0 +1,11 @@
+  readonly name: string;
+  readonly orientationDegrees: number;
+
+  constructor(id: string, name: string, orientationDegrees: number) {
```
Igual que el otro constructor.

oscar0310 replied:
Igual que el otro constructor.

oscar0310 commented on src/domain/entities/surf-conditions.ts:
```diff
@@ -0,0 +1,21 @@
+  readonly directionDegrees: number;
+
+  constructor(
+    id: string,
```
El constructor no aporta nada a la modelización del problema.

oscar0310 commented on src/domain/entities/surf-conditions.ts:
```diff
@@ -0,0 +1,21 @@
+export class SurfConditions {
+  readonly id: string;
```
El id de un objeto nunca va dentro del mismo objeto.

oscar0310 reviewed (commented):
Buenas @vpedrosa he estado viendo tu proyecto y me ha parecido muy interesantes,  pero creo que tiene algunos problemas que te quiero comentar.

Los issues son problemas , por eso los issues como Representación de x  no sirven te están dando solución directamente . Tienes que partir de la idea de expresar lo que tienes y lo que quieres hacer , e ir desgranando el problema hasta llegar a algo simple y tu código es la solución.  Teniendo en cuenta esto  hay issues en los que pones se necesitan estructuras, ahí ya estas dando la solución y no sería valido. 

Añadirte que JJ nos explicó que los identificadores nunca iban dentro del objeto, si no fuera para identificarlos y hacerlos únicos.

Ánimo , creo que con estos conceptos lo vas a sacar fácil.

vpedrosa committed:
refactor: simplifica SurfConditions a tipo de datos del dominio

Se elimina el id interno y el constructor. El id no es un atributo
del dominio, sino un mecanismo de identificación que pertenece a la
capa de almacenamiento (M1). La SSOT se garantizará usando el id
como clave externa de un repositorio.

Se usa type en lugar de class para representar solo la estructura
de datos sin lógica.

Closes #56

vpedrosa committed:
refactor: simplifica SurfSpot a tipo de datos del dominio

Se elimina el id interno y el constructor. El id no es un atributo
del dominio (el README describe el spot por nombre y orientación),
sino un mecanismo de identificación que pertenece a la capa de
almacenamiento (M1).

Se usa type en lugar de class para representar solo la estructura
de datos sin lógica.

Closes #57

vpedrosa commented on src/domain/entities/surf-spot.ts:
```diff
@@ -0,0 +1,11 @@
+export class SurfSpot {
+  readonly id: string;
```
> He eliminado los identificadores internos de las entidades. El id se gestionará como clave externa en M1.

Ídem.

vpedrosa commented:
Gracias por la revisión @oscar0310. He corregido ambos puntos: eliminados los ids internos y constructores, ahora las entidades son tipos de datos puros. ¿Cómo lo ves ahora?

oscar0310 reviewed (commented):
buenas @vpedrosa el código en si lo veo bien pero el problema principales que no has seguido una metodología correcta al crear los issues para llegar a la solución, me refiero a que los issues como representación de x no son validos, tienes que partir de la idea como te expliqué ayer de lo que tengo y lo que quieres hacer  e ir desgranando el problema principal en problemas pequeños hasta llegar al código de la solución. En si este objetivo como no explicó jj en clase es mucho más importante ese proceso que el código.  Además no me queda claro como garantizas la única fuente de verdad porque veo dos entidades  una para los spot y otra para las condiciones,  Supongo que la idea es mostrar para cada spot las condiciones en  una fecha, pero ¿Cómo te aseguras para el spot no haya varias condiciones en la misma fecha? .

vpedrosa commented:
Muchas gracias por el review @oscar0310.
Partiendo de lo que me dices, como los issues deben:
  - Plantear problemas del dominio, no tareas o soluciones
  - Partir de "lo que tengo" y "lo que quiero hacer"
  - Desgranar el problema principal en problemas pequeños
 
 Me he puesto a identificar los issues que estarían mal (por ejemplo el #45 "Representación de spot de surf con orientación": esto efectivamente es una tarea/solución, no un problema). Y he llegado a la conclusión de que los issues 42-51, el 53 y del 55-57 no siguen la metodología correcta. Para arreglarlo, voy a hacer el proceso desde el principio, intentando no sesgarme con el código y la forma de trabajo que ya tenía. Como tampoco tiene sentido mantener los issues 54 y 52, voy a realizar el proceso desde el principio de nuevo, pero con la información que me habéis dado entre todos los revisores. También me he fijado en cómo ha resuelto el objetivo @jacarmona364 en el PR https://github.com/GabrielFranciscoSM/practicas-IV/pull/9.
 
 Dicho eso cierro los issues que ya estaban resueltos, y replanteo el problema, pero ahora siguiendo bien el proceso.

vpedrosa committed:
refactor: elimina modelo de dominio para reestructurar con metodología correcta

Se elimina el código existente porque los issues no seguían la metodología
correcta: estaban planteados como tareas/soluciones en lugar de problemas
del dominio. Se regenerará siguiendo el flujo:

1. Issue como problema del dominio (derivado de HU1)
2. Código como solución que cierra el issue

vpedrosa committed:
feat: define datos del spot de surf

closes #58

vpedrosa committed:
feat: define datos de condiciones del oleaje

closes #59

vpedrosa committed:
feat: garantiza SSOT para spots y condiciones

closes #60

vpedrosa committed:
fix: actualiza iv.yaml para apuntar a SpotData

vpedrosa commented:
> buenas @vpedrosa el código en si lo veo bien pero el problema principales que no has seguido una metodología correcta al crear los issues para llegar a la solución, me refiero a que los issues como representación de x no son validos, tienes que partir de la idea como te expliqué ayer de lo que tengo y lo que quieres hacer e ir desgranando el problema principal en problemas pequeños hasta llegar al código de la solución. En si este objetivo como no explicó jj en clase es mucho más importante ese proceso que el código. Además no me queda claro como garantizas la única fuente de verdad porque veo dos entidades una para los spot y otra para las condiciones, Supongo que la idea es mostrar para cada spot las condiciones en una fecha, pero ¿Cómo te aseguras para el spot no haya varias condiciones en la misma fecha? .

Si puedes echarle un vistazo te lo agradezco, creo que ahora contemplo todo.

JJ reviewed (commented):
Esta es la tercera revisión, a efectos de lo indicado en la guía docente: a partir de ahora, sólo se harán revisiones adicionales si *realmente* se está avanzando y mejorando el código y no simplemente cambiándolo, con ayuda externa o no.
Si no se ve una mejora *real* en la siguiente iteración, se considerará no superado este objetivo. 
Te recuerdo así mismo que la fecha de superación de este objetivo *en convocatoria extraordinaria* es mañana, por lo que cualquier modificación que hagas, hazlo a la mayor brevedad.

vpedrosa committed:
refactor: Modifico .gitignore para que no incluya .claude.

vpedrosa commented on .gitignore:
```diff
@@ -1,208 +1,33 @@
+.nyc_output/
+
+# Claude
+.claude/
```
Había copiado el .gitignore, no lo había escrito yo. Fallo mío.

vpedrosa replied:
De acuerdo, lo haré así si me vuelve a pasar en el futuro.

vpedrosa committed:
refactor: Modifico asegurando SSOT.
Modifico y aúno toda la información en una única tupla dentro de un Map.

Closes #60.

vpedrosa committed:
fix: Resuelvo error de SSOT con Date
Resuelvo un error. Date se asocia por referencia por lo que no asegura SSOT #60.

vpedrosa commented:
@JJ solicito revisión, por favor.

chelunike reviewed (approved):
Creo que se modeliza el problema perfecamente 👍

oscar0310 commented on tsconfig.json:
El archivo de configuración no forma parte de este objetivo.

oscar0310 reviewed (approved):
Ya veo bien el objetivo aunque te he dejado un comentario sobre tsconfig.json

JJ reviewed (approved):
👍
